at_client/lib/src/encryption_service/abstract_atkey_encryption.dart

```diff
@@ -44,13 +44,22 @@ abstract class AbstractAtKeyEncryption implements AtKeyEncryption {
       // to cache the encryptedSharedKey.
       await _notifyEncryptedSharedKey(atKey, encryptedSharedKey);
       // Store the sharedKey for future retrieval.
-      // Encrypt the sharedKey with currentAtSignPrivateKey and store it for future use.
+      // Encrypt the sharedKey with currentAtSignPublicKey and store it for future use.
+      String? currentAtSignEncryptionPublicKey;
+      try {
+        currentAtSignEncryptionPublicKey = await AtClientManager.getInstance()
```

Comment:
if publickey is not found in local.should we lookup remote?

Reply:
The encryption public key of current atsign gets from the .atKeys files and then gets synced to remote secondary. So in ideal scenario, the key is not available in local is highly unlikely.